File: web-app/src/main/resources/public/js/angular/user/user.service.js
(function() {
	
	'use strict';
	
	angular
		.module('user.module')
		.factory('UserService', UserService);
	
	UserService.$inject = ['$http', 'EndpointService'];
	
	function UserService($http, endpointService) {

		let USER_REST_ENDPOINT = endpointService.getUserEndpoints().REST_ENDPOINT;
		
		return {
				getAllUsers: getAllUsers,
				getUserById: getUserById,
		};
		
		function getAllUsers() {
			return $http.get(USER_REST_ENDPOINT);
		}
		
		function getUserById(userId) {
			return $http.get(USER_REST_ENDPOINT + userId);
		}
		
	}
	
})();

File: microservices/user/user-provider/src/main/java/com/sbm/user/service/UserService.java
package com.sbm.user.service;

import java.util.List;

import com.sbm.user.dao.UserDao;
import com.sbm.user.model.User;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class UserService {

	@Autowired
	private UserDao userDao;
	
	@Transactional
	public List<? extends User> getAllUsers() {
		return userDao.getAllUsers();
	}
	
	@Transactional
	public User getUserById(Long userId) {
		return userDao.getUserById(userId);
	}
	
}
